client/src/common/components/Filter.jsx

```diff
@@ -71,6 +71,7 @@ class Filter extends PureComponent {
   render() {
     const { query } = this.state;
     const { buttonContent, clearFilterButton, placeholder } = this.props;
+    const isClearButtonVisible = clearFilterButton && buttonContent && query;
```

Comment:
Test case - `query = " "` 😉

Reply:
When the user enters "  " the 'X' will appear. But as we discussed this is good solution because maybe the user accidentally entered that white-space and with the 'X' button visible the user can easily clear the input.